openeo/rest/datacube.py

```diff
@@ -1829,6 +1829,75 @@ def validate(self) -> List[dict]:
     def tiled_viewing_service(self, type: str, **kwargs) -> Service:
         return self._connection.create_service(self.flat_graph(), type=type, **kwargs)
 
+    def _get_spatial_extent_from_load_collection(self):
+        pg = self.flat_graph()
+        for node in pg:
+            if pg[node]["process_id"] == "load_collection":
+                if "spatial_extent" in pg[node]["arguments"] and all(
+                    cd
+                    for cd in pg[node]["arguments"]["spatial_extent"]
+                    for cd in ["east", "west", "south", "north"]
```

Comment:
There is something off here: `cd` is used in two for loops at the same time

I think the `all()` will always resolve to True

Reply:
maybe this was meant?
```
all(
    cd in pg[node]["arguments"]["spatial_extent"] 
    for cd in ["east", "west", "south", "north"]
)
```